docs/design_proposals/channel_config.md

```diff
@@ -0,0 +1,176 @@
+# Channel configuration file
+
+In corporate environments, you often run into the situation where you have to configure custom mirrors for channels like `conda-forge`.
+
+This can be done by creating a file in one of the following locations:
+
+1. `$PWD/rattler.toml`
+2. `$PWD/.rattler.toml`
+3. `$XDG_CONFIG_HOME/rattler/config.toml` (see [dirs crate](https://crates.io/crates/dirs))
+4. `$HOME/.config/rattler/config.toml`
+5. `$HOME/rattler.toml`
+6. `$HOME/.rattler.toml`
+
+If you want to use a different configuration location, you can override it with the `RATTLER_CHANNEL_CONFIG` environment variable.
```

Comment:
Maybe `RATTLER_CONFIG` instead of `RATTLER_CHANNEL_CONFIG` if we want to specify different behavior as well in this file at some point.